### PR TITLE
Update version numbers and add C++ header guards

### PR DESCRIPTION
**Pull request by yourslab**
- Update version numbers from v3.0.1 to v3.0.2
- Add C++ header guards

### CODE REVIEW

**Review by archigup — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,7 @@
 # Change Log for corePKCS11 Library
 
 ## v3.0.2 (July 2021)
+* [#118](https://github.com/FreeRTOS/corePKCS11/pull/118) Update version numbers and add C++ header guards
```
Comment (aggarw13) [inline CHANGELOG.md:4]:
```suggestion
* [#118](https://github.com/FreeRTOS/corePKCS11/pull/118) Add support for using library in C++ applications```

**Comment by archigup** [comment]
Might be good to add the new commit to changelog, and the other PR if we are merging that.

**Review by n9wxu — APPROVED** [review]

**Comment by yourslab** [comment]
> Might be good to add the new commit to changelog, and the other PR if we are merging that.

Yup I've added that already

File: CHANGELOG.md (hunk 1 above)
Comment (yourslab) [reply CHANGELOG.md:4]: I don't think this change is necessarily required to use the library in C++ applications

**Pull request merged**